### PR TITLE
Fix typos newly found by codespell

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Subject: fix typos

### Feature or Bugfix

- Bugfix

### CODE REVIEW

File: sphinx/ext/autosummary/__init__.py
Comment (picnixz) [inline sphinx/ext/autosummary/__init__.py:806]: For this one, you need to check the localized strings as well otherwise we could have other surprises such as #11689 (though, since it's a logger message it shouldn't affect people that much, but better be safe).

Comment (DimitriPapadopoulos) [reply sphinx/ext/autosummary/__init__.py:806]: My wrong, I thought the localisation files are updated separately by an "out-of-band" process. I have updated all localisation files.

**Review by picnixz — APPROVED** [review]
Looks good to me (still you'll need to wait for Adam's approval).

File: doc/index.rst
Comment (AA-Turner) [inline doc/index.rst:24]: `re-use` is fine

File: doc/latex.rst
Comment (AA-Turner) [inline doc/latex.rst:487]: As is `re-usable`

File: sphinx/locale/ar/LC_MESSAGES/sphinx.po
Comment (AA-Turner) [inline sphinx/locale/ar/LC_MESSAGES/sphinx.po:1]: All of the ``.po`` files can be excluded from the spelling checker, they're automatically updated from the source.

File: sphinx/writers/text.py
Comment (AA-Turner) [inline sphinx/writers/text.py]: `co-ordinates` is fine

**Review by AA-Turner — CHANGES_REQUESTED** [review]

File: sphinx/locale/ar/LC_MESSAGES/sphinx.po
Comment (DimitriPapadopoulos) [reply sphinx/locale/ar/LC_MESSAGES/sphinx.po:1]: That's what I thought, but I had been told otherwise in https://github.com/sphinx-doc/sphinx/pull/11735#discussion_r1370490500.

Any way, these updates are in a separate commit, they would be easy to discard.

File: doc/index.rst
Comment (DimitriPapadopoulos) [reply doc/index.rst:24]: It depends on the definition of fine. The rationale follows:
  * All majors dictionaries suggest `reuse` only:
    * [OED](https://www.oed.com/search/dictionary/?q=re-use)
    * [Cambridge](https://www.oxfordlearnersdictionaries.com/definition/english/reuse_1)
    * [Collins](https://www.collinsdictionary.com/dictionary/english/reuse)
    * [Longman](https://www.ldoceonline.com/dictionary/reuse)
    * [Merriam-Webster](https://www.merriam-webster.com/dictionary/reuse)
    * [Oxford Learner's Dictionary](https://dictionary.cambridge.org/dictionary/english/reuse)
  * From [The Blue Book of Grammar and Punctuation](https://www.grammarbook.com/blog/hyphens/hyphens-with-the-prefix-re/):
    > Use the hyphen with the prefix _re_ only when _re_ means _again_ AND omitting the hyphen would cause confusion with another word.
* Both `re-use` and `reuse` can be seen as correct, but:
  * The [Google Ngram Viewer](https://books.google.com/ngrams/graph?content=re-use%2Creuse&year_start=1800&year_end=2019&corpus=en-2019) shows `reuse`  wider use (not by much I must admit).

I do understand you have your own preference, but for a documentation that targets a large audience, what's wrong with following dictionaries, grammar rules and common use?

File: doc/latex.rst
Comment (DimitriPapadopoulos) [reply doc/latex.rst:487]: See above.

File: sphinx/writers/text.py
Comment (DimitriPapadopoulos) [reply sphinx/writers/text.py]: Both `co-ordinate` and `coordinate` are fine, especially in British English according to dictionaries. However, the [Google Ngram Viewer](https://books.google.com/ngrams/graph?content=co-ordinate%2Ccoordinate&year_start=1800&year_end=2019&corpus=en-2019) shows `coordinate`  in wider use, including in the [British English corpus](https://books.google.com/ngrams/graph?content=co-ordinate%2Ccoordinate&year_start=1800&year_end=2019&corpus=en-GB-2019) strangely enough.

It's up to you, but what's wrong with following common use when targetting a large and international audience?

File: sphinx/locale/ar/LC_MESSAGES/sphinx.po
Comment (picnixz) [reply sphinx/locale/ar/LC_MESSAGES/sphinx.po:1]: If we change a localized string, I think what is autogenerated is an empty translation, hence we probably need to manually update translations in this case but I may be wrong.

File: sphinx/writers/text.py
Comment (picnixz) [reply sphinx/writers/text.py]: Even though co-ordinate is fine, I personally prefer one without "-". Also it removes one character from the code base so it's technically an optimization!

Comment (hugovk) [reply sphinx/writers/text.py]: How about a compromise and use "coördinates" like the New Yorker! 🙃

https://www.newyorker.com/culture/culture-desk/the-curse-of-the-diaeresis

**Review by AA-Turner — APPROVED** [review]

File: doc/index.rst
Comment (AA-Turner) [reply doc/index.rst:24]: Thanks for the explanation, I'm content to adopt your recommendation.

**Pull request merged**
